hoot-core/src/main/cpp/hoot/core/io/GeoNamesReader.cpp

```diff
@@ -110,6 +151,89 @@ void GeoNamesReader::open(const QString& url)
   {
     throw HootException("Error opening file for reading: " + url);
   }
+
+  // Now figure out what type of Geonames file we have
+  QString line = QString::fromUtf8(_fp.readLine().constData());
+  QStringList fields;
+  fields.append(line.split('\t'));
+
+//  if (_mil_columns.size() == fields.size())
+  if (fields.size() == 36)
+  {
+    _GEONAMESID = 1;  // UFI Field
+    _LATITUDE = 3;
+    _LONGITUDE = 4;
+
+    // Geonames.nga.mil column names
+    // https://geonames.nga.mil/gns/html/gis_countryfiles.html
+    _columns << "region_font_code"; // RC
```

Comment:
Is there any chance the order of these can change in the future? If so, should be made configurable.

Reply:
I don't think the format will change. The column order is what they post on their websites.

Reply:
Ok, then don't worry about doing that.